lib/level_db.rb

```diff
@@ -113,6 +113,24 @@ def each(options={}, &block)
     def snapshot
       Snapshot.new(@db)
     end
+
+    def compact_range(options={})
+      from = options[:from]
+      to = options[:to]
+      @db.compact_range(encode_key(from), encode_key(to))
+    end
+
+    def full_compaction
+      sstables = @db.get_property(SSTABLES_PROPERTY)
+      pre_compaction_sstables = nil
+      until pre_compaction_sstables == sstables
+        pre_compaction_sstables = sstables
+        compact_range
+        sstables = @db.get_property(SSTABLES_PROPERTY)
+      end
```

Comment:
Are you sure this can't end up in an infinite loop? What if other threads manipulate the database at the same time? Or would that just make this loop go another go, and then stop?